File: tasks/task3/Prime.rb
def is_6k_1(num)
    return false if num <= 1
    return true if num <= 3
    return false if num % 2 == 0 || num % 3 == 0
  
    i = 5
    while i * i <= num
      return false if num % i == 0 || num % (i + 2) == 0
      i += 6
    end
  
    true
  end
  
  print 'Enter a number: '
  n = gets.chomp.to_i
  puts "Prime till #{n} are:"
  (2..n).each { |i| puts i if is_6k_1(i) }
  

File: tasks/task3/prime.hs
is6k1 :: Int -> Bool
is6k1 num
    | num <= 1 = False
    | num <= 3 = True
    | num `mod` 2 == 0 || num `mod` 3 == 0 = False
    | otherwise = doCheck num 5
  where
    doCheck num' i
        | i * i > num' = True
        | num' `mod` i == 0 || num' `mod` (i + 2) == 0 = False
        | otherwise = doCheck num' (i + 6)

main :: IO ()
main = do
    putStrLn "Enter a number: "
    input <- getLine
    let n = read input :: Int
    putStrLn $ "Prime till " ++ show n ++ " are:"
    mapM_ (print) [x | x <- [2..n], is6k1 x]
